external/st_audiofile/src/st_audiofile.c

```diff
@@ -478,6 +544,28 @@ uint64_t st_read_f32(st_audio_file* af, float* buffer, uint64_t count)
     case st_audio_file_mp3:
         count = drmp3_read_pcm_frames_f32(af->mp3, count, buffer);
         break;
+    case st_audio_file_wv:
+        if (af->cache.wv.mode & MODE_FLOAT) {
+            count = WavpackUnpackSamples(af->wv, (int32_t*)buffer, (uint32_t)count);
+        } else {
+            uint32_t channels = af->cache.wv.channels;
+            int32_t* buf_i32 = (int32_t*)malloc(4 * channels * count);
+            if (!buf_i32) {
+                return 0;
+            }
+            count = channels * WavpackUnpackSamples(af->wv, buf_i32, (uint32_t)count);
+            if (!(af->cache.wv.mode & MODE_FLOAT)) {
+                if (af->cache.wv.bitrate < 32) {
+                    int d = 32 - af->cache.wv.bitrate;
+                    for (uint64_t i = 0; i < count; i++) {
+                        buf_i32[i] <<= d;
```

Comment:
So basically in wavpack if you use integers less than 32 bit length, you're not gaining any space?

Reply:
Acoording to a [comment](https://github.com/dbry/WavPack/blob/ffc72be679dcb13b7e6004060c4e7baf5ba24401/src/unpack_utils.c#L94) in `WavpackUnpackSamples()`, the function always returns 32-bit data, padded when needed. So optimizing sfizz to use less memory when dealing with < 32 bit WV audio does not seem possible.

Reply:
OK np, in memory we store as float anyway. It was just out of curiosity.